### ISSUE TITLE
D9: Error: Cannot instantiate interface Drush\SiteAlias\AliasRecord

### ISSUE THREAD

**Issue by kenorb** [drush-ops/drush#3649]
**Describe the bug**

After installing Drush 9.3.0 via Composer, it doesn't work. It fails with the error:

> Fatal error: Uncaught Error: Cannot instantiate interface Drush\SiteAlias\AliasRecord

**To Reproduce**

```
# mkdir drush9 && cd drush9/
# composer require drush/drush:^9.3
...
# composer show | grep drush
drush/drush                           9.3.0   Drush ...
# composer exec -v drush
> __exec_command: drush

Fatal error: Uncaught Error: Cannot instantiate interface Drush\SiteAlias\AliasRecord in /var/www/html/drush9/vendor/drush/drush/src/SiteAlias/SiteAlia
sManager.php on line 25

Error: Cannot instantiate interface Drush\SiteAlias\AliasRecord in /var/www/html/drush9/vendor/drush/drush/src/SiteAlias/SiteAliasManager.php on line 2
5

Call Stack:
    0.2014     394224   1. {main}() /var/www/html/drush9/vendor/drush/drush/drush:0
    0.2049     401896   2. require('/var/www/html/drush9/vendor/drush/drush/drush.php') /var/www/html/drush9/vendor/drush/drush/drush:4
    0.3838    1358712   3. Drush\Runtime\Runtime->run() /var/www/html/drush9/vendor/drush/drush/drush.php:66
    0.3838    1358712   4. Drush\Runtime\Runtime->doRun() /var/www/html/drush9/vendor/drush/drush/src/Runtime/Runtime.php:41
    0.3838    1358712   5. Drush\Preflight\Preflight->preflight() /var/www/html/drush9/vendor/drush/drush/src/Runtime/Runtime.php:58
    0.5580    2768776   6. Drush\SiteAlias\SiteAliasManager->__construct() /var/www/html/drush9/vendor/drush/drush/src/Preflight/Preflight.php:258
# ./vendor/bin/drush
... (same error as above) ...
# php --version
PHP 7.2.8 (cli) (built: Jul 21 2018 07:52:06) ( NTS )
```

**Expected behavior**

It not errors.

**Actual behavior**

It errors.

**Workaround**

Use Drush 8.

### System Configuration

| Q               | A
| --------------- | ---
| Drush version?  | 9.3.0
| Drupal version? | 8.x
| PHP version     | 7.2
| OS?             | Linux Docker container under Windows

**Additional information**

None.

**Comment by greg-1-anderson** [comment]
Hm, I can't remember a time when AliasRecord was ever an interface, but perhaps it was.

I guess what's happening here is that you somehow have some mismatch related to the fact that the Site Alias code was migrated into a separate project. Maybe try removing composer.lock and vendor and re-install fresh, and see if that works better.

**Comment by greg-1-anderson** [comment]
I see that AliasRecord is an interface in the consolidation/site-alias project. This might be a mistake.

I can't reproduce this, though; it works for me when I followed your instructions. Are you sure that your drush9 directory started out empty, and you haven't mixed anything else in with your Drush install?

**Comment by greg-1-anderson** [comment]
I was able to reproduce this by taking a local git clone of Drush and then running `git checkout 9.3.0`.

`composer install` did not fix this problem, but `composer update` did.

For now I'm writing this problem off as a bad install; it should not happen if you upgrade Drush cleanly. Let me know if you reproduce this with a clean install (e.g. mixing Drush 9 dev-master with an earlier version of Drush installed as a site-local Drush in a Drupal 8 site, etc.)

**Comment by greg-1-anderson** [comment]
Even though I do not feel this is a supported configuration, I fixed it in the latest dev-master, 1844496e.

Make sure that you run `composer update` after `git pull` if you update via git.

**Comment by greg-1-anderson** [comment]
Note that if you have a site-local Drush 8, you cannot use Drush 9.  Drush 9 will *always* redispatch to the site-local Drush 8.  It could not do otherwise, or it would crash due to dependency hell problems, perhaps with symptoms similar to the above.

If you are having problems such as the above with a redispatch from a global Drush 9 to a site-local Drush 8, first make sure that the two versions of Drush are not intermixed in any way (not in the same vendor). If you get anything other than a correct run on the site-local Drush 8, that's a bug. Please specify that you are using a site-local Drush when reporting such bugs.

If you wish to "try out" Drush 9 commands, you must either a) use separate Drupal sites, or b) use a global Drush 8 and a global Drush 9 (installed in separate `vendor` directories).
